### DISCUSSION TITLE
Maintain BaseModel key order in `model_json_schema()`?

### DISCUSSION THREAD

**Discussion by john-sungjin** [pydantic/pydantic#5460] [Question] [answered]
Is there a way to preserve the order of the BaseModel's attributes in the `properties` key of the output JSON schema? For example, given this BaseModel:

```
class Test(BaseModel):
    c: list[str]
    a: str
    b: int
```

I'd like `Test.model_json_schema()` to list the properties in the order [c, b, a], but instead the properties are sorted in alphabetical order:

```
{'properties': {'a': {'title': 'A', 'type': 'string'},
                'b': {'title': 'B', 'type': 'integer'},
                'c': {'items': {'type': 'string'},
                      'title': 'C',
                      'type': 'array'}},
 'required': ['c', 'a', 'b'],
 'title': 'Test',
 'type': 'object'}
```

They're correctly ordered in 'required', but I'd rather have the 'properties' sorted correctly since optional fields wouldn't show up in this list.

**Comment by dmontagu** [comment]
When I run the following code:
```python
import json

from pydantic import BaseModel


class Test(BaseModel):
    c: list[str]
    a: str
    b: int


print(json.dumps(Test.model_json_schema(), indent=2))
```
I get 
```
{
  "title": "Test",
  "type": "object",
  "properties": {
    "c": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "title": "C"
    },
    "a": {
      "type": "string",
      "title": "A"
    },
    "b": {
      "type": "integer",
      "title": "B"
    }
  },
  "required": [
    "c",
    "a",
    "b"
  ]
}
```
(The fields are in the same order in the JSON schema as in the class definition.)

If you are getting different behavior, could you share info about the version of pydantic you are using (i.e., share the results of `from pydantic.version import version_info; print(version_info())`, and provide a self-contained script that misbehaves?

A brief look at the code makes me think it should always be properly preserving the field order, but it's possible I've missed something. It would definitely help if I could reproduce the issue though.

**Comment by john-sungjin** [comment, marked as answer]
Oh whoops - I was using pprint to display the schemas and it looks like that was reorganizing the objects. Thank you for taking a look!
